.github/scripts/manage-vm.py

```diff
@@ -144,7 +155,49 @@ def get_runner_token(github_repo_owner, github_repo, github_token):
         raise ValueError(f"Failed to get runner registration token: {result}")
 
 
-def create_vm(sdk, args):
+def wait_for_runner_registration(
+    client: Github,
+    vm_id: str,
+    github_repo_owner: str,
+    github_repo: str,
+    sleep_interval: int = 10,
+    intervals: int = 60,
+) -> bool:
+    logger.info(
+        "Waiting for runner registration for %d*%d seconds", intervals, sleep_interval
+    )
+
+    for i in range(intervals):
```

Comment:
If you want to wait for a specific event to occur, the preferred solution would be:
```
finished_at = time.monotonic() + intervals
while True:
    if find_runner_by_name(...):
        return True
    if time.monotonic() > finished_at:
         return False
    time.sleep(max(min(sleep_interval, finished_at - time.monotonic()), 0))
```

Reply:
But if you do not care much for sleeping a bit more, then bob's your uncle, the for approach works fine.

Reply:
I thought about that, but for me more important to have specific number of retries, than to wait some amount of time.